### ISSUE TITLE
Cannot read SSH RSA key pair (Android 9+)

### ISSUE THREAD

**Issue by JackieKu** [iiordanov/remote-desktop-clients#200]
version: v4.0.8
Pixel 4 with 2020-04 update
Lowering the target API level should help:
https://android-developers.googleblog.com/2018/03/cryptography-changes-in-android-p.html

The key is indeed without a passphrase.
```
04-11 22:59:05.696 12907 14391 I PubkeyUtils: Passphrase empty, recovering directly.
04-11 22:59:05.697 12907 14391 I PubkeyUtils: Either key is not encrypted and we were given passphrase, or the passphrase is wrong,or the key is corrupt.
04-11 22:59:05.697 12907 14391 W System.err: java.security.NoSuchAlgorithmException: The BC provider no longer provides an implementation for KeyFactory.RSA.  Please see https://android-developers.googleblog.com/2018/03/cryptography-changes-in-android-p.html for more details.
04-11 22:59:05.698 12907 14391 W System.err:    at sun.security.jca.Providers.checkBouncyCastleDeprecation(Providers.java:563)
04-11 22:59:05.698 12907 14391 W System.err:    at sun.security.jca.Providers.checkBouncyCastleDeprecation(Providers.java:330)
04-11 22:59:05.698 12907 14391 W System.err:    at java.security.KeyFactory.getInstance(KeyFactory.java:235)
04-11 22:59:05.698 12907 14391 W System.err:    at com.iiordanov.pubkeygenerator.PubkeyUtils.recoverKeyPair(PubkeyUtils.java:172)
04-11 22:59:05.698 12907 14391 W System.err:    at com.iiordanov.pubkeygenerator.PubkeyUtils.decryptAndRecoverKeyPair(PubkeyUtils.java:241)
04-11 22:59:05.698 12907 14391 W System.err:    at com.iiordanov.bVNC.SSHConnection.decryptAndRecoverKey(SSHConnection.java:398)
04-11 22:59:05.698 12907 14391 W System.err:    at com.iiordanov.bVNC.SSHConnection.authenticateWithPubKey(SSHConnection.java:411)
04-11 22:59:05.698 12907 14391 W System.err:    at com.iiordanov.bVNC.SSHConnection.initializeSSHTunnel(SSHConnection.java:172)
04-11 22:59:05.698 12907 14391 W System.err:    at com.iiordanov.bVNC.RemoteCanvas.getPort(RemoteCanvas.java:661)
04-11 22:59:05.698 12907 14391 W System.err:    at com.iiordanov.bVNC.RemoteCanvas.startVncConnection(RemoteCanvas.java:460)
04-11 22:59:05.698 12907 14391 W System.err:    at com.iiordanov.bVNC.RemoteCanvas.access$200(RemoteCanvas.java:95)
04-11 22:59:05.698 12907 14391 W System.err:    at com.iiordanov.bVNC.RemoteCanvas$2.run(RemoteCanvas.java:289)
04-11 22:59:05.699 12907 14391 W System.err: java.lang.Exception: Failed to decrypt key-pair. Please ensure you've entered your passphrase correctly in the 'SSH Passphrase' field on the main screen.
04-11 22:59:05.699 12907 14391 W System.err:    at com.iiordanov.bVNC.SSHConnection.decryptAndRecoverKey(SSHConnection.java:400)
04-11 22:59:05.699 12907 14391 W System.err:    at com.iiordanov.bVNC.SSHConnection.authenticateWithPubKey(SSHConnection.java:411)
04-11 22:59:05.699 12907 14391 W System.err:    at com.iiordanov.bVNC.SSHConnection.initializeSSHTunnel(SSHConnection.java:172)
04-11 22:59:05.699 12907 14391 W System.err:    at com.iiordanov.bVNC.RemoteCanvas.getPort(RemoteCanvas.java:661)
04-11 22:59:05.699 12907 14391 W System.err:    at com.iiordanov.bVNC.RemoteCanvas.startVncConnection(RemoteCanvas.java:460)
04-11 22:59:05.699 12907 14391 W System.err:    at com.iiordanov.bVNC.RemoteCanvas.access$200(RemoteCanvas.java:95)
04-11 22:59:05.699 12907 14391 W System.err:    at com.iiordanov.bVNC.RemoteCanvas$2.run(RemoteCanvas.java:289)
```

**Comment by iiordanov** [comment]
For your immediate needs, you can downgrade by taking an older version from here: https://github.com/iiordanov/remote-desktop-clients/releases

I may not be able to target a lower API as the minimum target API is 28 for Google Play at the moment, but I'll see if there is a workaround.

**Comment by iiordanov** [comment]
There is a fix in beta (v4.0.9) that I will fast-track to production now.

Thanks for your report!
iordan
